File: Projects/proj2A/proj2A.py
"""
Project 2A - Python Factor Counting
"""

for i in range(2, 1000):
    the_number = i
    num_factors = 0
    for j in range(2, i-1):
        if (the_number % j == 0):
            num_factors +=1
            while (the_number % j == 0):
                the_number = the_number / j
    if num_factors == 3:
        print(f"{i} has three factors")

File: Projects/proj2A/proj2A.c
/*
CS 212 - proj2A : Factor counter
Sabrina Zhang 

*/


#include <stdio.h>

int main(){

    for (int i = 2; i < 1000; i++)
    {
        int the_number = i;
        int num_factors = 0;
        for (int j = 2; j < i - 1; j++)
        {
            if (the_number % j == 0)
            {
                num_factors++;
            
                while (the_number % j == 0)
                {
                    the_number = the_number / j;
                }
            }    
        }

        if (num_factors == 3)
        {
            printf("%d has three factors\n", i);
    
        }
    }

}
